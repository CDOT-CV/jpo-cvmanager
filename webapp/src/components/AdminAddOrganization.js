import React, { useState } from "react";
import { Form } from "react-bootstrap";
import { useForm } from "react-hook-form";
import EnvironmentVars from "../EnvironmentVars";

import "../components/css/Admin.css";
import "react-widgets/styles.css";

const AdminAddOrganization = (props) => {
  const [successMsg, setSuccessMsg] = useState("");
  const [errorState, setErrorState] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm();

  const sendPostData = async (json) => {
    if (props.isLoginActive()) {
      props.setLoading(true);
      setErrorState(false);

      try {
        const res = await fetch(EnvironmentVars.adminAddOrg, {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
            Authorization: props.authLoginData["token"],
          },
          body: JSON.stringify(json),
        });

        const status = res.status;
        const data = await res.json();
        props.setLoading(false);
        if (status === 200) {
          setSuccessMsg("Organization Creation is successful.");
          reset();
          resetMsg();
          props.updateOrganizationData();
        } else if (status === 400) {
          setErrorMessage(data.message);
          setErrorState(true);
          resetMsg();
        } else if (status === 500) {
          setErrorMessage(data.message);
          setErrorState(true);
        }
      } catch (exception_var) {
        props.setLoading(false);
        setErrorState(true);
        setErrorMessage(String(exception_var));
        console.error(exception_var);
      }
    }
  };

  const onSubmit = (data) => {
    sendPostData(data);
  };

  function resetMsg() {
    setTimeout(() => setSuccessMsg(""), 5000);
  }

  return (
    <div>
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Group className="mb-3" controlId="name">
          <Form.Label>Organization Name</Form.Label>
          <Form.Control
            type="text"
            placeholder="Enter organization name"
            {...register("name", {
              required: "Please enter the organization name",
            })}
          />
          {errors.name && <p className="errorMsg">{errors.name.message}</p>}
        </Form.Group>

        {successMsg && <p className="success-msg">{successMsg}</p>}
        {errorState && (
          <p className="error-msg">
            Failed to add organization due to error: {errorMessage}
          </p>
        )}
        <div className="form-control">
          <label></label>
          <button type="submit" className="admin-button">Add Organization</button>
        </div>
      </Form>
    </div>
  );
};

export default AdminAddOrganization;
