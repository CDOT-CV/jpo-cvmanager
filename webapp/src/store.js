import { configureStore } from "@reduxjs/toolkit";
import rsuReducer from "./slices/rsuSlice";
import userReducer from "./slices/userSlice";
import wzdxReducer from "./slices/wzdxSlice";
import configReducer from "./slices/configSlice";

const store = configureStore({
  reducer: {
    rsu: rsuReducer,
    user: userReducer,
    wzdx: wzdxReducer,
    config: configReducer,
  },
});

export default store;
