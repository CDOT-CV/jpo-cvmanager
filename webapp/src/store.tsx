import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import intersectionMapReducer from './components/map/map-slice'
import intersectionMapLayerStyleReducer from './components/map/map-layer-style-slice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
    },
    preloadedState,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
