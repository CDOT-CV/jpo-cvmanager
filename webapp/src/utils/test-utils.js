/**
 * This function searches for the every react-aria SSR ids in a given HTMLElement node and replace every attribute values with a static id
 *
 * This can be usefull when you're trying to generate a snapshot of components using react-aria under the hood
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceReactAriaIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: "class",
      updateFunc: (val) =>
        val.replace(
          /css-[0-9a-z]{6}-MuiTableCell-root-MuiTablePagination-root/g,
          "css-mocked-MuiTableCell-root-MuiTablePagination-root"
        ),
    },
  ];

  container.querySelectorAll("td").forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)));
      }
    });
  });
  container.querySelectorAll("input[aria-invalid]").forEach((item) => {
    delete item["aria-invalid"];
  });
  return container;
}

export { replaceChaoticIds };
