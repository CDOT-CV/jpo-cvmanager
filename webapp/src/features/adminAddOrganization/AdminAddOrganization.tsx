import React from 'react'
import { Form } from 'react-bootstrap'
import { useForm } from 'react-hook-form'
import {
  // actions
  addOrg,
} from './adminAddOrganizationSlice'
import { useSelector, useDispatch } from 'react-redux'
import toast from 'react-hot-toast'

import '../adminRsuTab/Admin.css'
import 'react-widgets/styles.css'
import { AnyAction, ThunkDispatch } from '@reduxjs/toolkit'
import { RootState } from '../../store'

export type AdminAddOrgForm = {
  name: string
  email: string
}

const AdminAddOrganization = () => {
  const dispatch: ThunkDispatch<RootState, void, AnyAction> = useDispatch()
  const notifySuccess = (message: string) => toast.success(message)
  const notifyError = (message: string) => toast.error(message)
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<AdminAddOrgForm>()

  const onSubmit = (data: AdminAddOrgForm) => {
    dispatch(addOrg({ json: data })).then((data: any) => {
      data.payload.success
        ? notifySuccess(data.payload.message)
        : notifyError('Failed to add organization due to error: ' + data.payload.message)
    })
  }

  return (
    <div>
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Group className="mb-3" controlId="name">
          <Form.Label>Organization Name</Form.Label>
          <Form.Control
            type="text"
            placeholder="Enter organization name"
            {...register('name', {
              required: 'Please enter the organization name',
            })}
          />
          <Form.Label>Organization Email</Form.Label>
          <Form.Control
            type="text"
            placeholder="Enter organization email"
            {...register('email', {
              required: 'Please enter the organization email',
            })}
          />
          {errors.name && (
            <p className="errorMsg" role="alert">
              {errors.name.message}
            </p>
          )}
        </Form.Group>

        <div className="form-control">
          <label></label>
          <button type="submit" className="admin-button">
            Add Organization
          </button>
        </div>
      </Form>
    </div>
  )
}

export default AdminAddOrganization
