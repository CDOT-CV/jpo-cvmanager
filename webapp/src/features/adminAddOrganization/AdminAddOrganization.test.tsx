import React from 'react'
import { render } from '@testing-library/react'
import AdminAddOrganization from './AdminAddOrganization'
import { Provider } from 'react-redux'
import { setupStore } from '../../store'
import { replaceChaoticIds } from '../../utils/test-utils'
import { BrowserRouter, Route, Routes } from 'react-router-dom'

it('should take a snapshot', () => {
  const { container } = render(
    <Provider store={setupStore({})}>
      <BrowserRouter>
        <Routes>
          <Route path="*" element={<AdminAddOrganization />} />
        </Routes>
      </BrowserRouter>
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
