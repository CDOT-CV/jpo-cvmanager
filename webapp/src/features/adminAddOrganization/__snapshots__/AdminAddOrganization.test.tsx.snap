// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <form
      class=""
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="name"
        >
          Organization Name
        </label>
        <input
          class="form-control"
          id="name"
          name="name"
          placeholder="Enter organization name"
          type="text"
        />
        <label
          class="form-label"
          for="name"
        >
          Organization Email
        </label>
        <input
          class="form-control"
          id="name"
          name="email"
          placeholder="Enter organization email"
          type="text"
        />
      </div>
      <div
        class="form-control"
      >
        <label />
        <button
          class="admin-button"
          type="submit"
        >
          Add Organization
        </button>
      </div>
    </form>
  </div>
</div>
`;
